tests/tests_geomstats/test_landmarks.py

```diff
@@ -1,164 +1,214 @@
-        self.landmarks_b = landmark_set_b
-        self.landmarks_c = landmark_set_c
-
-    def test_belongs(self):
-        result = self.space_landmarks_in_sphere_2d.belongs(self.landmarks_a)
-        expected = True
-        self.assertAllClose(result, expected)
-
-    def test_belongs_vectorization(self):
-        landmark_sets = gs.array([self.landmarks_a, self.landmarks_b])
-        result = self.space_landmarks_in_sphere_2d.belongs(landmark_sets)
-        expected = gs.array([True, True])
-        self.assertAllClose(result, expected)
-
-    def test_l2_metric_log_and_squared_norm_and_dist(self):
-        """Test that squared norm of logarithm is squared dist."""
-        tangent_vec = self.l2_metric_s2.log(
-            point=self.landmarks_b, base_point=self.landmarks_a
-        )
-        log_ab = tangent_vec
-        result = self.l2_metric_s2.squared_norm(
-            vector=log_ab, base_point=self.landmarks_a
-        )
-        expected = self.l2_metric_s2.dist(self.landmarks_a, self.landmarks_b) ** 2
-
-        self.assertAllClose(result, expected)
-
-    def test_l2_metric_log_and_exp(self):
-        """Test that exp and log are inverse maps."""
-        tangent_vec = self.l2_metric_s2.log(
-            point=self.landmarks_b, base_point=self.landmarks_a
-        )
-        result = self.l2_metric_s2.exp(
-            tangent_vec=tangent_vec, base_point=self.landmarks_a
-        )
-        expected = self.landmarks_b
-
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_tf_only
-    def test_l2_metric_inner_product_vectorization(self):
-        """Test the vectorization inner_product."""
-        n_samples = self.n_landmark_sets
-        landmarks_ab = self.l2_metric_s2.geodesic(self.landmarks_a, self.landmarks_b)
-        landmarks_bc = self.l2_metric_s2.geodesic(self.landmarks_b, self.landmarks_c)
-        landmarks_ab = landmarks_ab(self.times)
-        landmarks_bc = landmarks_bc(self.times)
-
-        tangent_vecs = self.l2_metric_s2.log(
-            point=landmarks_bc, base_point=landmarks_ab
-        )
-
-        result = self.l2_metric_s2.inner_product(
-            tangent_vecs, tangent_vecs, landmarks_ab
-        )
-
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-    @geomstats.tests.np_autograd_and_tf_only
-    def test_l2_metric_dist_vectorization(self):
-        """Test the vectorization of dist."""
-        n_samples = self.n_landmark_sets
-        landmarks_ab = self.l2_metric_s2.geodesic(self.landmarks_a, self.landmarks_b)
-        landmarks_bc = self.l2_metric_s2.geodesic(self.landmarks_b, self.landmarks_c)
-        landmarks_ab = landmarks_ab(self.times)
-        landmarks_bc = landmarks_bc(self.times)
-
-        result = self.l2_metric_s2.dist(landmarks_ab, landmarks_bc)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-    @geomstats.tests.np_autograd_and_tf_only
-    def test_l2_metric_exp_vectorization(self):
```

Comment:
This test had an exp whose shape was correct, but exp_shape is skipped with
```
skip_test_exp_shape = True
```

Please add this test back, since we do not want to decrease protection.

Reply:
#1386 #1385 is the reason for this error. unfortunately I cannot skip this. instead I will add the test back.